osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850SetScheduleCommand.java

```diff
@@ -282,61 +315,67 @@ private void writeScheduleEntryForRelay(final DeviceMessageLog deviceMessageLog,
 
         for (final ScheduleEntryDto schedule : scheduleList) {
             for (final LightValueDto lightValue : schedule.getLightValue()) {
+                this.setScheduleEntry(ssld, ssldDataService, relaySchedulesEntries, relayType, settings, schedule,
+                        lightValue);
+            }
+        }
 
-                final List<Integer> indexes = new ArrayList<>();
+        return relaySchedulesEntries;
+    }
 
-                if (lightValue.getIndex() == 0
-                        && (RelayType.TARIFF.equals(relayType) || RelayType.TARIFF_REVERSED.equals(relayType))) {
+    private ScheduleEntry setScheduleEntry(final Ssld ssld, final SsldDataService ssldDataService,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 7 parameters, which is greater than 6 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00107)